src/test/java/org/junitpioneer/internal/PioneerPathAssert.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.internal;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.singletonList;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+
+import org.assertj.core.api.PathAssert;
+
+public class PioneerPathAssert extends PathAssert {
+
+	public static PioneerPathAssert assertThatPath(Path actual) {
+		return new PioneerPathAssert(actual);
+	}
+
+	private PioneerPathAssert(Path path) {
+		super(path);
+	}
+
+	public PioneerPathAssert canAddTextFile() {
+		canAddTextFileInternal();
+		return this;
+	}
+
+	public PioneerPathAssert canReadTextFile() {
```

Comment:
I prefer the name `canReadAndWriteFile`. This is a bit long, but the assertion _does_ write a file and that can fail on its own, so it needs to clear that the assertion does that. Same argument for no "Text" as above.